src/lib/Libwin/win_remote_shell.c

```diff
@@ -114,26 +115,30 @@ disconnect_close_pipe(HANDLE hpipe)
  * @retval	!0 - error number, describing error while trying to create std pipes
  *
  */
-int
+DWORD
 create_std_pipes(STARTUPINFO* psi, char *pipename_append, int is_interactive)
 {
 	char stdoutpipe[PIPENAME_MAX_LENGTH] = "";
 	char stderrpipe[PIPENAME_MAX_LENGTH] = "";
 	char stdinpipe[PIPENAME_MAX_LENGTH] = "";
 	SECURITY_ATTRIBUTES SecAttrib = {0};
 	SECURITY_DESCRIPTOR SecDesc;
-	int err = 0;
+	DWORD err = 0;
+	char logbuff[LOG_BUF_SIZE] = { '\0' };
 
 	if (psi == NULL || pipename_append == NULL)
-		return -1;
+		return ERROR_INVALID_PARAMETER;
 
 	/* Assign NULL DACL to the scurity descriptor, allowing all access to the pipe */
 	if (InitializeSecurityDescriptor(&SecDesc, SECURITY_DESCRIPTOR_REVISION) == 0) {
 		err = GetLastError();
+		log_err(-1, __func__, "Failed to initialize a security descriptor");
 		return err;
 	}
 	if (SetSecurityDescriptorDacl(&SecDesc, TRUE, NULL, FALSE) == 0) {
 		err = GetLastError();
+		sprintf(logbuff, "Failed to set DACL for security descriptor with error %lu", err);
+		log_err(-1, __func__, logbuff);
 		return err;
 	}
 	SecAttrib.nLength = sizeof(SECURITY_ATTRIBUTES);
```

Comment:
Please add error logging here:
https://github.com/PBSPro/pbspro/blob/bb1fa7be4315449eb94bb63e280b260c4cec3c0f/src/lib/Libwin/win_remote_shell.c#L177

Reply:
Same for CreateNamedPipe
https://github.com/PBSPro/pbspro/blob/bb1fa7be4315449eb94bb63e280b260c4cec3c0f/src/lib/Libwin/win_remote_shell.c#L204

Reply:
I did not add log message here because we are returing the error and logging it in the calling function. But added it here too now to address the comments.

Reply:
I think logically it is necessary here, because the calling functions are not logging the error.
Please look pbs_open_demux.c